### ISSUE TITLE
Access denied from ECR even though EC2 have read access

### ISSUE THREAD

**Issue by darwin67** [concourse/registry-image-resource#18]
Hi!

Just want to thank you for all the awesome work first!

The issue I've encountered is, it seems like `registry-image` can't grab images from ECR even though the IAM of the EC2 have read access to it.

Here's the screenshot of the error.
![401-ecr](https://user-images.githubusercontent.com/5746693/53939394-c0163200-4067-11e9-867c-37d17646d459.png)

I tried digging through the logs to find that error but was unsuccessful so the screenshot is all I can provide.

**Comment by vito** [comment]
Right - we haven't yet added support for any auth mechanisms except for basic auth.

I would like to avoid using EC2 IAM roles directly from the resource for this - https://github.com/concourse/concourse/issues/3023 gives some context as to why. tl;dr: we want resources to receive all credential config explicitly from `source:` and would like to support EC2 IAM roles by representing it as a credential manager.

At the end of the day this is probably blocked on that issue. Once it's done I suppose we would just need to allow configuring this resource with an access key ID/secret, which would come from the IAM credential manager.

For now I would recommend sticking with the `docker-image` resource - since this is the new and shiny resource that's meant to learn from all the mistakes we made in the past, we're trying to keep it "pure" until we're ready to tell everyone to make the switch.

> Just want to thank you for all the awesome work first!

Thanks! :D

**Comment by darwin67** [comment]
@vito Thanks for clarifying!

Do you want to close this issue then or keep it open as reference?

**Comment by vito** [comment]
@darwin67 np! Let's keep this open since we'll still need to implement the params in this resource to support ECR auth (by way of an access key/secret in `source`), even after that issue is done.

**Comment by tvon** [comment]
I am in a similar boat using Azure Container Registry, not sure if that is worth a separate issue.
